src/navigation/navigation-events.js

```diff
@@ -136,7 +128,24 @@ function createPopStateEvent(state, originalMethodName) {
   return evt;
 }
 
-if (isInBrowser) {
+let historyApiIsPatched = false;
+
+// We patch the history API so single-spa is notified of all calls to pushState/replaceState.
+// We patch addEventListener/removeEventListener so we can capture all popstate/hashchange event listeners,
+// and delay calling them until single-spa has finished mounting/unmounting applications
+export function patchHistoryApi() {
```

Comment:
Could we add `urlRerouteOnly` parameter into this function and pass it to patchedUpdateState? It is still need to call start while we need the features that urlRerouteOnly true did.

Reply:
👍🏿  good idea, i have done so